packages/shared/lib/contexts/onboarding/actions/claimShimmerRewards.ts

```diff
@@ -0,0 +1,62 @@
+import { get } from 'svelte/store'
+
+import { localize } from '@core/i18n'
+import { DEFAULT_TRANSACTION_OPTIONS, getOutputOptions } from '@core/wallet'
+import { showAppNotification } from '@lib/notifications'
+import { MILLISECONDS_PER_SECOND } from '@lib/time'
+import { sleep } from '@lib/utils'
+
+import { prepareShimmerClaimingAccount } from '../helpers'
+import { IShimmerClaimingAccount } from '../interfaces'
+import { onboardingProfile, updateShimmerClaimingAccounts } from '../stores'
+
+export async function claimShimmerRewards(): Promise<void> {
+    try {
+        const shimmerClaimingAccounts = get(onboardingProfile)?.shimmerClaimingAccounts
+        const unclaimedShimmerClaimingAccounts =
+            shimmerClaimingAccounts?.filter((shimmerClaimingAccount) => shimmerClaimingAccount?.unclaimedRewards > 0) ??
```

Comment:
It is probably better to use `(shimmerClaimingAccount) => shimmerClaimingAccount?.state === ShimmerAccountState.Unclaimed` instead since it's more readable and leaves the logic for what unclaimed means in one place. But we'd most likely have to split the state into two: `UnclaimedWithRewards` and `UnclaimedWithoutRewards`. :man_shrugging: